### DISCUSSION TITLE
Adding 3D Building to MapFileTileSource

### DISCUSSION THREAD

**Discussion by ahxmed991** [mapsforge/vtm#920] [Q&A] [answered]
Hello, i wanted to code something for my university project. I am digging the codes for a week and couldn't find a way. I am trying to build a 3d building (or 3d line) in MapFileTileSource and want to change Latitude, Longitude (location), color, length, width, height in realtime. I tried implementing ITileDataSource to MultiMapFileTileSource because i saw in ExtrusionsTest but because of the setBaseMap method, those didn't work together.

**Comment by devemux86** [comment, marked as answer]
The 3D buildings are read from the vector maps, like the Mapsforge map files.
Their rendering can be changed with the [render themes](https://github.com/mapsforge/vtm/blob/master/docs/Rendertheme.md).

> setBaseMap method

If you study the `setBaseMap` method, you can write it differently:
```java
VectorTileLayer vtl = new OsmTileLayer(mMap, tts);
mMap.layers().add(vtl);
```

**Reply by ahxmed991** [reply]
<img width="1026" alt="Ekran Resmi 2022-04-17 06 04 14" src="https://user-images.githubusercontent.com/103830880/163698514-b13334f3-94e5-467a-b1f4-e70b54d0cff8.png">
I want the map to look like this but in this system I can't specify the latitude and longitude of the 3d bars. I also don't want the bars to repeat like a tile.
